packages/proxy/lib/http/request-middleware.ts

```diff
@@ -111,15 +113,18 @@ const EndRequestsToBlockedHosts: RequestMiddleware = function () {
 }
 
 const StripUnsupportedAcceptEncoding: RequestMiddleware = function () {
-  // Cypress can only support plaintext or gzip, so make sure we don't request anything else
-  const acceptEncoding = this.req.headers['accept-encoding']
-
-  if (acceptEncoding) {
-    if (acceptEncoding.includes('gzip')) {
-      this.req.headers['accept-encoding'] = 'gzip'
-    } else {
-      delete this.req.headers['accept-encoding']
-    }
+  // ignore as header is a string
+  // @ts-ignore
+  const acceptEncoding: string = this.req.headers['accept-encoding'] || ''
+  const acceptEncodingArray: string[] = acceptEncoding
+  .split(',')
+  .filter((str) => AllowedContentEncodings.includes(str))
+  .map((str) => str.trim())
```

Comment:
this filtering is not 100% correct, see the MDN doc for accept-encoding: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept-Encoding

the quality value (`q`) is ignored in this parsing. `q` should probably be preserved, so filtering on `/^(gzip|br|deflate)/.test(str)` or something would be more correct. also, lines 121 and 122 ought to be reversed, right now `gzip ` (note the space) would be filtered out by this logic.